bindata/network/ovn-kubernetes/managed/ovnkube-control-plane.yaml

```diff
@@ -33,6 +33,7 @@ spec:
       annotations:
         hypershift.openshift.io/release-image: {{.ReleaseImage}}
         target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+        network.operator.openshift.io/ovnkube-script-lib-hash: "{{.OVNKubeConfigHash}}"
```

Comment:
So this is interesting because now control plane might rollout together with node for config changes that have nothing to do with it.

I can't think of an actual problem right now but this may indeed bring them about.

In general I think that the parameters should be set as env variables in the specific pod specs and that script lib should just use those env variables; instead of the using the template variables directly from the script lib. This is however out of scope and a job for a different PR.

Tagging @kyrtapz for awareness and to check if he has ideas on how this can be a problem.

Reply:
Although undesirable, I don't think it would be an actual problem. The same would happen when for example the ovn-k image changes during an upgrade.

Reply:
> So this is interesting because now control plane might rollout together with node for config changes that have nothing to do with it.
> I can't think of an actual problem right now but this may indeed bring them about.

So `OVNKubeConfigHash` contains the hash value of `ovnkube-script-lib` configmap that is only used in ovnkube-node. I think that introducing this here is confusing. 
Additionally, this would cause unnecessary rollouts of  ovnkube-control-plane every time a  ovnkube-node specific setting changes.

> In general I think that the parameters should be set as env variables in the specific pod specs and that script lib should just use those env variables; instead of the using template variables directly from the script lib. This is however out of scope and a job for a different PR.

Using env vars sounds reasonable but I would avoid introducing yet another way of rolling out changes just for the purpose of this PR.

Reply:
> So `OVNKubeConfigHash` contains the hash value of `ovnkube-script-lib` configmap that is only used in ovnkube-node. I think that introducing this here is confusing. Additionally, this would cause unnecessary rollouts of ovnkube-control-plane every time a ovnkube-node specific setting changes.

I think using `ovnkube-script-lib` for ovn-control-plane also is reasonable. As I mention, the equivalent consideration would be when the ovn-k image changes on upgrade.

> Using env vars sounds reasonable but I would avoid introducing yet another way of rolling out changes just for the purpose of this PR.

Unsure what you mean with `another way`. I guess I am talking about **replacing** one way (using configuration template variables directly in script lib) with another (using env variables) so that we don't trigger rollouts of unnecessary components when certain configuration parameters are changed. Of course, it might be unreasonable to make the change in one go so we might need to assume doing this progressively.